### PR TITLE
New seg release dinov2b release

### PR DESCRIPTION
**Pull request by probicheaux**
## What does this PR do?

Releases new RF-DETR sizes:
Object detection:
L, XL, XXL
Segmentation:
N, S, M, L, XL, XXL

This PR also changes the default License to a hybrid License: Apache 2.0 for everything except the code and weights related to Obj-det XL and XXL, which are under the Platform Model License -- the License is clarified in the code.

## Type of Change

- New feature (non-breaking change that adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Testing

- [x] I have tested this change locally
- [ ] I have added/updated tests for this change

**Test details:**

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [ x I have updated the documentation accordingly (if applicable)

## Additional Context

This PR renames the old RF-DETR-Large -> RF-DETR-LargeDeprecated

### CODE REVIEW

File: rfdetr/datasets/coco_eval.py
[the remark this answers is not in the document]
Comment (jrdalenberg) [reply rfdetr/datasets/coco_eval.py]: Suggestion:

Change `rfdetr/engine.py`, line 265:
```python
coco_evaluator = CocoEvaluator(base_ds, iou_types, args)
```

Change `CocoEvaluator` into:

```python
class CocoEvaluator(object):
    def __init__(self, coco_gt, iou_types, args=None, max_dets=500):
        assert isinstance(iou_types, (list, tuple))
        coco_gt = copy.deepcopy(coco_gt)
        self.coco_gt = coco_gt

        if 'eval_max_dets' in args:
            self.max_dets = args.eval_max_dets
        else:
            self.max_dets = max_dets

        self.iou_types = iou_types
        self.coco_eval = {}
        for iou_type in iou_types:
            self.coco_eval[iou_type] = COCOeval(coco_gt, iouType=iou_type)
            self.coco_eval[iou_type].params.maxDets = [1, 10, self.max_dets]

        self.img_ids = []
        self.eval_imgs = {k: [] for k in iou_types}
```

[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/datasets/coco_eval.py]: thanks for catching this!

File: rfdetr/__init__.py [hunk 1]
```diff
@@ -9,4 +9,23 @@
 if os.environ.get("PYTORCH_ENABLE_MPS_FALLBACK") is None:
     os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"
 
-from rfdetr.detr import RFDETRBase, RFDETRLarge, RFDETRNano, RFDETRSmall, RFDETRMedium, RFDETRSegPreview
+from rfdetr.platform.models import (
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/__init__.py]: resolved; renamed

File: rfdetr/__init__.py [hunk 2]
```diff
@@ -9,4 +9,23 @@
...
+    RFDETRXLargeConfig,
```
Comment (Borda) [inline rfdetr/__init__.py]:
```suggestion
    RFDETRXLarge,
```
likely a typo :)

File: rfdetr/platform/models.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    dec_n_points: int = 4
+    num_windows: int = 1
+    patch_size: int = 20
+    projector_scale: List[Literal["P4",]] = ["P4"]
+    out_feature_indexes: List[int] = [3, 6, 9, 12]
+    num_classes: int = 365
+    positional_encoding_size: int = 700 // 20
+    resolution: int = 700
+    pretrain_weights: str = "rf-detr-xlarge.pth"
+    license: str = "PML-1.0"
```
Comment (Borda) [inline rfdetr/platform/models.py]: not needed now, but we may rather use StrEnum to maintain consitency and prevent typos

File: rfdetr/config.py [hunk 4]
```diff
@@ -37,6 +37,7 @@ class ModelConfig(BaseModel):
     cls_loss_coef: float = 1.0
     segmentation_head: bool = False
     mask_downsample_ratio: int = 4
+    license: str = "apache-2.0"
```
Comment (Borda) [inline rfdetr/config.py]:
```suggestion
    license: str = "Apache-2.0"
```

File: rfdetr/detr.py [hunk 5]
```diff
@@ -468,10 +465,115 @@ def get_model_config(self, **kwargs):
...
+        return TrainConfig(**kwargs)
+
+class RFDETRLarge(RFDETR):
+    size = "rfdetr-large"
+    def __init__(self, **kwargs):
+        self.init_error = None
+        self.is_deprecated = False
+        try:
+            super().__init__(**kwargs)
+        except Exception as e:
```
Comment (Borda) [inline rfdetr/detr.py]: Does the failing loading emit a specific exception, or use that then shield any eny even unrelated exception...

File: rfdetr/detr.py
Comment (jrdalenberg) [inline rfdetr/detr.py]: In line 326 of `detr.py`, variable `predictions` is overwritten. As a result, the if-statement in line 330 is never triggered and segmentation models do not have any masks as output during inference.

File: rfdetr/datasets/coco_eval.py [hunk 6]
```diff
@@ -269,3 +271,83 @@ def evaluate(self):
...
+        stats[2] = _summarize(1, iouThr=.75, maxDets=self.params.maxDets[2])
+        stats[3] = _summarize(1, areaRng='small', maxDets=self.params.maxDets[2])
+        stats[4] = _summarize(1, areaRng='medium', maxDets=self.params.maxDets[2])
+        stats[5] = _summarize(1, areaRng='large', maxDets=self.params.maxDets[2])
+        stats[6] = _summarize(0, maxDets=self.params.maxDets[0])
+        stats[7] = _summarize(0, maxDets=self.params.maxDets[1])
+        stats[8] = _summarize(0, maxDets=self.params.maxDets[2])
+        stats[9] = _summarize(0, areaRng='small', maxDets=self.params.maxDets[2])
+        stats[10] = _summarize(0, areaRng='medium', maxDets=self.params.maxDets[2])
+        stats[11] = _summarize(0, areaRng='large', maxDets=self.params.maxDets[2])
```
Comment (Borda) [inline rfdetr/datasets/coco_eval.py]: I may place it as simple list, this makes me think/validate if the numbers are in order

File: rfdetr/models/segmentation_head.py [hunk 7]
```diff
@@ -104,6 +104,41 @@ def forward(self, spatial_features: torch.Tensor, query_features: list[torch.Ten
...
+        # query features: [(B, N, C)] for each decoder layer
+        # output: dict containing the intermediate results
+        target_size = (image_size[0] // self.downsample_ratio, image_size[1] // self.downsample_ratio)
+        spatial_features = F.interpolate(spatial_features, size=target_size, mode='bilinear', align_corners=False)
+
+        num_points = max(spatial_features.shape[-2], spatial_features.shape[-2] * spatial_features.shape[-1] // 16)
+
+        output_dicts = []
+
+        if not skip_blocks:
```
Comment (Borda) [inline rfdetr/models/segmentation_head.py]:
```suggestion
        if skip_blocks:
```
I would for readbility rather have without `not`

File: rfdetr/detr.py [hunk 8]
```diff
@@ -468,10 +465,115 @@ def get_model_config(self, **kwargs):
...
         return TrainConfig(**kwargs)
 
+
+class RFDETRLargeNew(RFDETR):
+    size = "rfdetr-large"
+    def get_model_config(self, **kwargs):
+        return RFDETRLargeConfig(**kwargs)
+
+    def get_train_config(self, **kwargs):
+        return TrainConfig(**kwargs)
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/detr.py]: leaving just in case someone wants to be sure they get the new model

File: rfdetr/detr.py [hunk 9]
```diff
@@ -25,14 +25,21 @@
 
 from rfdetr.config import (
     RFDETRBaseConfig,
-    RFDETRLargeConfig,
+    RFDETRLargeDeprecatedConfig,
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/detr.py]: Deprecated is more descriptive, leaving as is

File: rfdetr/main.py [hunk 10]
```diff
@@ -62,8 +64,18 @@
     "rf-detr-small.pth": "https://storage.googleapis.com/rfdetr/small_coco/checkpoint_best_regular.pth",
     "rf-detr-medium.pth": "https://storage.googleapis.com/rfdetr/medium_coco/checkpoint_best_regular.pth",
     "rf-detr-seg-preview.pt": "https://storage.googleapis.com/rfdetr/rf-detr-seg-preview.pt",
+    "rf-detr-large-edge.pth": "https://storage.googleapis.com/rfdetr/rf-detr-large-edge.pth",
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/main.py]: moving to `rf-detr-large-2026.pth` so we download a new file on someones machine if they already have the old

File: rfdetr/platform/models.py [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    pretrain_weights: str = "rf-detr-xxlarge.pth"
+    license: str = "PML-1.0"
+
+
+class RFDETRXLarge(RFDETR):
+    size = "rfdetr-xlarge"
+
+    def __init__(self, accept_platform_model_license: bool = False, **kwargs):
+        if not accept_platform_model_license is True:
+            raise ValueError(
```
[the remark this answers is not in the document]
Comment (probicheaux) [reply rfdetr/platform/models.py]: I talked to Brad, he ok'd this as is. API key would be more restrictive I guess?

File: rfdetr/models/segmentation_head.py (hunk 7 above)
Comment (probicheaux) [reply rfdetr/models/segmentation_head.py]: seems ok as is, can update later
